packages/ai/src/core.ts

```diff
@@ -0,0 +1,114 @@
+import type { Client } from "edgedb";
+import type { ResolvedConnectConfig } from "edgedb/dist/conUtils";
+
+import type { AIOptions, QueryContext, RAGRequest } from "./types.js";
+
+export function createAI(client: Client, options: Partial<AIOptions> = {}) {
+  return new EdgeDBAI(client, options);
+}
+
+export class EdgeDBAI {
+  /** @internal */
+  private readonly baseUrl: Promise<string>;
+  private readonly options: AIOptions;
+  private readonly context: QueryContext;
+
+  /** @internal */
+  constructor(
+    public readonly client: Client,
+    options: Partial<AIOptions> = {},
+    context: Partial<QueryContext> = {}
+  ) {
+    this.baseUrl = EdgeDBAI.getBaseUrl(client);
+    this.options = {
+      model: options.model ?? "gpt-3.5-turbo",
+      prompt: options.prompt ?? {
+        name: "builtin::rag-default",
+      },
```

Comment:
I'm not sure we need these defaults. The prompt is optional anyway, as the server will pick a default prompt. And the model is dependent on which providers you have configured, so it doesn't seem safe to use an openai model as the default?

Reply:
Oh, that's a good point. Will fix!